fvm/meter/meter.go

```diff
@@ -1,553 +1,52 @@
 	return MeterParameters{
-		computationLimit:        uint64(umax) << MeterExecutionInternalPrecisionBytes,
-		memoryLimit:             math.MaxUint64,
-		computationWeights:      DefaultComputationWeights,
-		memoryWeights:           DefaultMemoryWeights,
-		storageInteractionLimit: math.MaxUint64,
-		eventEmitByteLimit:      math.MaxUint64,
+		ComputationMeterParameters: DefaultComputationMeterParameters(),
+		MemoryMeterParameters:      DefaultMemoryParameters(),
+		EventMeterParameters:       DefaultEventMeterParameters(),
+		InteractionMeterParameters: DefaultInteractionMeterParameters(),
 	}
 }
 
-func (params MeterParameters) WithComputationLimit(limit uint) MeterParameters {
-	newParams := params
-	newParams.computationLimit = uint64(limit) << MeterExecutionInternalPrecisionBytes
-	return newParams
-}
-
-func (params MeterParameters) WithComputationWeights(
-	weights ExecutionEffortWeights,
-) MeterParameters {
-	newParams := params
-	newParams.computationWeights = weights
-	return newParams
-}
-
-func (params MeterParameters) WithMemoryLimit(limit uint64) MeterParameters {
-	newParams := params
-	newParams.memoryLimit = limit
-	return newParams
-}
-
-func (params MeterParameters) WithMemoryWeights(
-	weights ExecutionMemoryWeights,
-) MeterParameters {
-	newParams := params
-	newParams.memoryWeights = weights
-	return newParams
-}
-
-func (params MeterParameters) WithStorageInteractionLimit(
-	maxStorageInteractionLimit uint64,
-) MeterParameters {
-	newParams := params
-	newParams.storageInteractionLimit = maxStorageInteractionLimit
-	return newParams
-}
-
-func (params MeterParameters) WithEventEmitByteLimit(
-	byteLimit uint64,
-) MeterParameters {
-	newParams := params
-	newParams.eventEmitByteLimit = byteLimit
-	return newParams
-}
-
-func (params MeterParameters) ComputationWeights() ExecutionEffortWeights {
-	return params.computationWeights
-}
-
-// TotalComputationLimit returns the total computation limit
-func (params MeterParameters) TotalComputationLimit() uint {
-	return uint(params.computationLimit >> MeterExecutionInternalPrecisionBytes)
-}
-
-func (params MeterParameters) MemoryWeights() ExecutionMemoryWeights {
-	return params.memoryWeights
-}
-
-// TotalMemoryLimit returns the total memory limit
-func (params MeterParameters) TotalMemoryLimit() uint64 {
-	return params.memoryLimit
-}
-
 // Meter collects memory and computation usage and enforces limits
 // for any each memory/computation usage call it sums intensity multiplied by the weight of the intensity to the total
 // memory/computation usage metrics and returns error if limits are not met.
 type Meter struct {
 	MeterParameters
 
-	computationUsed uint64
-	memoryEstimate  uint64
-
-	computationIntensities MeteredComputationIntensities
-	memoryIntensities      MeteredMemoryIntensities
-
-	storageUpdateSizeMap     map[StorageInteractionKey]uint64
-	totalStorageBytesRead    uint64
-	totalStorageBytesWritten uint64
-
-	totalEmittedEventBytes uint64
+	*MemoryMeter
```

Comment:
btw, do these need to be pointer?  maybe just have `New<blah>Meter` return struct instead of pointer